### PR TITLE
Separate CIS and USG in the client

### PR DESCRIPTION
**Pull request by renanrodrigo**
## Why is this needed?
This PR solves all of our problems because it implements the changes needed to turn USG into an actual resource on the client side. This aligns client behaviour with the implementation done for the backend

## Test Steps
kept behavior:
- enabling CIS should work on X, B
- enabling USG should work on J
- both should work on F, but USG gets enabled in both cases , with a message telling about the transition

new behavior:
- upgrading from older versions of pro client where CIS was enabled as USG on F, J should enable the actual USG resource

Integration tests TBD
---

- [x] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: uaclient/cli/enable.py [hunk 1]
```diff
@@ -449,6 +447,13 @@ def action_enable(args, *, cfg, **kwargs) -> int:
     enabled_services = _enabled_services(cfg).enabled_services
     all_dependencies = _dependencies(cfg).services
 
+    # On Focal, CIS transitioned to USG.
+    # If a user tries to enable CIS on Focal, they should get USG instead.
+    if "cis" in names and system.get_release_info().series == "focal":
+        entitlements_found.append("usg")
+        entitlements_not_found.remove("cis")
+        interactive_only_print(messages.CIS_IS_NOW_USG)
```
Comment (lucasmoura) [inline uaclient/cli/enable.py:455]: I wonder if this logic should sit on the entitlements module instead. The rationale is that this should also apply to the enable API, which uses the `entitlement_factory` function. I know that this function is not used here, but I think the entitlement module should be the one aware of this situation, and not the CLI module

**Review by orndorffgrant — COMMENTED** [review]
Looking good! A few little things.
I guess we can't really test yet until the resource is up on the backend

File: uaclient/cli/enable.py (hunk 1 above)
Comment (renanrodrigo) [reply uaclient/cli/enable.py:455]: The problem is the still somehow - as we need to do this during the enablement step, but not for other commands.
If we try to be always smart, the workaround will grow bigger than we want to maintain imho.
I'm leaning towards adding a similar snippet in the API call and live with it.

Comment (orndorffgrant) [reply uaclient/cli/enable.py:455]: I think that would be okay. I think ideally we would still install the cis packages on focal too, though, unless product has agreed that it's okay to break that?

Comment (renanrodrigo) [reply uaclient/cli/enable.py]: So I have changed the USG class to accomodate the message+packageset changes for CIS on Focal, and applied the logic for both enablement flows.

File: lib/postinst-migrations.sh [hunk 2]
```diff
@@ -81,4 +82,15 @@ if dpkg --compare-versions "$PREVIOUS_PKG_VER" lt "35~"; then
     if [ -f "/var/lib/ubuntu-advantage/private/machine-token.json" ]; then
         touch "/var/lib/ubuntu-advantage/marker-only-series-check"
     fi
+
+    # On Focal and Jammy, the CIS service transitioned to USG from v35
+    # This changes the lists and authentication files to represent that
+    # in case CIS is enabled
+    if [ "$VERSION_ID" = "20.04" ] || [ "$VERSION_ID" = "22.04" ]; then
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply lib/postinst-migrations.sh]: The only thing missing in this PR now is integration tests. I have manually tested it many times up to the point where we can without the backends, and will have the behave stuff in place asap

**Comment by renanrodrigo** [comment]
@orndorffgrant @lucasmoura this is done feature-wise, just missing the beave testing.
If you want to merge this to finalize the release code, I can send a separate PR with just the tests.

**Review by orndorffgrant — APPROVED** [review]
I'm okay merging this ahead of integration tests, but would like @lucasmoura to approve as well before we do

**Review by lucasmoura — APPROVED** [review]

File: uaclient/api/u/pro/services/enable/v1.py [hunk 3]
```diff
@@ -170,16 +171,27 @@ def _enable(
...
         raise exceptions.NotSupported()
 
+    # On Focal, CIS transitioned to USG.
+    # If a user tries to enable CIS on Focal, they should get USG instead.
+    if (
+        service_to_enable == "cis"
+        and system.get_release_info().series == "focal"
+    ):
+        service_to_enable = "usg"
+        USGEntitlement.cis_version = True
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply uaclient/api/u/pro/services/enable/v1.py:186]: I'm hate hating this for years now I hate this part a little less

**Pull request merged**